src/app/dim-ui/RichDestinyText.tsx

```diff
@@ -1,112 +1,105 @@
+  { unicode: '', icon: sidearm,      objectiveHash: rT['[Sidearm]']             },
+  { unicode: '', icon: gLauncherFF,  objectiveHash: rT['']                     },
   // Heavy Weapons
-  { objectiveHash: 2152699013, unicode: '', substring: null, icon: gLauncher    },
-  { objectiveHash: 2203404732, unicode: '', substring: null, icon: rLauncher    },
-  { objectiveHash: 1788114534, unicode: '', substring: null, icon: machinegun   },
-  { objectiveHash:  989767424, unicode: '', substring: null, icon: sword        },
+  { unicode: '', icon: gLauncher,    objectiveHash: rT['[Grenade Launcher]']    },
+  { unicode: '', icon: rLauncher,    objectiveHash: rT['[Rocket Launcher]']     },
+  { unicode: '', icon: machinegun,   objectiveHash: rT['[Machine Gun]']         },
+  { unicode: '', icon: sword,        objectiveHash: rT['[Sword]']               },
   // Artifacts that can be picked up and used as weapons
-  { objectiveHash: 4231452845, unicode: '', substring: null, icon: scorchCannon },
+  { unicode: '', icon: scorchCannon, objectiveHash: rT['']                     },
   // Gambit - Blockers
-  { objectiveHash:  276438067, unicode: '', substring: null, icon: smlBlocker   },
-  { objectiveHash: 3792840449, unicode: '', substring: null, icon: medBlocker   },
-  { objectiveHash: 2031240843, unicode: '', substring: null, icon: lrgBlocker   },
+  { unicode: '', icon: smlBlocker,   objectiveHash: rT['[Small Blocker]']       },
+  { unicode: '', icon: medBlocker,   objectiveHash: rT['[Medium Blocker]']      },
+  { unicode: '', icon: lrgBlocker,   objectiveHash: rT['[Large Blocker]']       },
   // Quest Markers
-  { objectiveHash: 3915460773, unicode: '', substring: null, icon: questMarker  },
+  { unicode: '', icon: questMarker,  objectiveHash: rT['[Quest]']               },
   // Breakers
-  { objectiveHash: 3068403538, unicode: '', substring: null, icon: overload     },
-  { objectiveHash: 2678922819, unicode: '', substring: null, icon: pierce       },
-  { objectiveHash: 3879088617, unicode: '', substring: null, icon: stagger      }
-];
+  { unicode: '', icon: overload,     objectiveHash: rT['[Disruption]']          },
+  { unicode: '', icon: pierce,       objectiveHash: rT['[Shield-Piercing]']     },
+  { unicode: '', icon: stagger,      objectiveHash: rT['[Stagger]']             }
+].map((c) => ({ ...c, substring: null }));
 
 /**
  * given defs, uses known examples from the manifest
  * and returns a localized string-to-icon conversion table
- *           "[Rocket Launcher]" -> <svg>
+ *           "[Rocket Launcher]" -> '<svg>'
  */
 const generateConversionTable = _.once((defs: D2ManifestDefinitions) => {
   // loop through conversionTable entries to update them with manifest string info
   baseConversionTable.forEach((iconEntry) => {
-    const objectiveDef = defs.Objective.get(iconEntry.objectiveHash);
+    const objectiveDef = iconEntry.objectiveHash && defs.Objective.get(iconEntry.objectiveHash);
```

Comment:
yeah I was trying to figure out how to do this in the table, did not even think about checking here